modules/collect/src/main/java/com/opengamma/collect/result/Result.java

```diff
@@ -466,6 +481,23 @@ public T getValue() {
   }
 
   /**
+   * Returns the actual result value if calculated successfully, or the specified
+   * default value if a failure occurred.
+   * <p>
+   * If this result is a success then the result value is returned.
+   * If this result is a failure then the default value is returned.
+   * <p>
+   * Application code is recommended to use {@link #map(Function)} and
+   * {@link #flatMap(Function)} in preference to this method.
+   *
+   * @param defaultValue  the default value to return if the result is a failure, may be null
+   * @return either the result value or the default value
+   */
+  public T getValueOrElse(@Nullable T defaultValue) {
```

Comment:
I'm not sure adding ``@Nullable`` is the right approach. I think consistency with ``Optional.orElse`` is less important than consistency with ``getValue`` which is on the same class and has almost the same name.

The intended usage is to provide a default value for a failure, and given that a successful result can't be null the variable you're assigning to shouldn't be nullable either.

The only use case I can think of is using null as a sentinel value to represent something that's missing, and that's not something we should encourage. That's basically what ``Result`` is for. And there's ``Optional`` in Java 8 too. 

Making it nullable sends a message that we think using null is OK when we don't.